compiler/src/dotty/tools/dotc/transform/SpecializeTuples.scala

```diff
@@ -37,15 +37,14 @@ class SpecializeTuples extends MiniPhase:
   end transformApply
 
   override def transformSelect(tree: Select)(using Context): Tree = tree match
-    case Select(qual, nme._1) if isAppliedSpecializableTuple(qual.tpe.widen) =>
-      Select(qual, nme._1.specializedName(qual.tpe.widen.argInfos.slice(0, 1)))
```

Comment:
The issue was that `qual.tpe.widen` returned `U[Int, Int]`, `isAppliedSpecializableTuple` detected the tuple type through internal aliasing, but `qual.tpe.widen.argInfos` used the args of `U[Int, Int]` instead of the args of its dealiased type `Tuple2[Unit, Unit]`